### ISSUE TITLE
Internal workings of 'visible' objects metadata.

### ISSUE THREAD

**Issue by soyeonm** [allenai/ai2thor#815]
Hello,

I thank you all the time for your kind and detailed replies.
I have a question regarding the inner workings of the 'visible' properties of objects.

![스크린샷 2021-07-11 오전 12 02 34](https://user-images.githubusercontent.com/77866067/125167419-52501700-e1db-11eb-902b-17745e025871.png)

Is the object considered visible if the minimum depth of the segmentation mask of a certain object ID is below or equal to the visibility distance? Or is there some other inner workings?

Due to some qualifications/ restrictions on a task, I am trying to approximately implement the visibility of an object without using metadata.

I am using AI2THOR 2.1.0.

Thank you very much all the time.

**Comment by mattdeitke** [comment]
Hi @soyeonm,

**TL;DR:** There is some other inner workings that allows us to compute visibility faster, but the segmentation approach described give the same answer as what is done internally (just slower).

> Is the object considered visible if the minimum depth of the segmentation mask of a certain object ID is below or equal to the visibility distance?

This is the ideal definition, where we check every single point on the object, and if any of them are within `visibilityDistance` of the agent's camera, then `visible` is set to `True`. However, running segmentation is quite slow, so we do not use this approach.

Instead, we approximate the above metric for efficiency purposes. Rather than checking the distance of every point in a segmentation mask, we annotate every single object with a **"visibility points"**, and we only check if any of these few visibility points are in the frame and within `visibilityDistance`.

![image](https://user-images.githubusercontent.com/28768645/125174563-0d949380-e17b-11eb-9473-701ace7fd565.png)

For instance, in the above image, the yellow spheres are the visibility points on the `Pot`. Then, to compute if it is visible, we only check if any of these are visible, instead of checking for every possible point on the object or in the segmentation frame.

That said, both solutions should give the same results :)

**Comment by soyeonm** [comment]
Thank you very much for your kind response!
